src/github/logic.py

```diff
@@ -296,6 +296,14 @@ def maybe_automerge_pull_request(pull_request: PullRequest) -> bool:
     return False
 
 
+def maybe_delete_branch_if_merged(pull_request: PullRequest):
+    if pull_request.merged():
```

Comment:
Just to note: even though this is called after `maybe_automerge`, the `pull_request` state at this point will not be merged if it's handling the same webhook event. This is because `maybe_automerge` doesn't (currently) update the state of the pull request object. Instead, it will fire on the next webhook event that gets triggered on the `close` action. Just an fyi while testing.

Reply:
yup, i think it makes sense that this will only happen on the PR close action.